langchain/agents/agent_toolkits/openapi/planner.py

```diff
@@ -0,0 +1,375 @@
+    name = "requests_get"
+    description = """Use this to GET content from a website.
+Input to the tool should be a json string with 2 keys: "url" and "output_instructions".
+The value of "url" should be a string. The value of "output_instructions" should be instructions on what information to extract from the response, for example the id(s) for a resource(s) that the GET request fetches.
+"""
+    response_length: Optional[int] = MAX_RESPONSE_LENGTH
+    llm_chain = llm_chain = LLMChain(
+        llm=DEFAULT_LLM,
+        prompt=PromptTemplate(
+            template="""Here is an API response:\n\n{response}\n\n====
+Your task is to extract some information according to these instructions: {instructions}
+When working with API objects, you should usually use ids over names.
+If the response indicates an error, you should instead output a summary of the error.
+
+Output:""",
+            input_variables=["response", "instructions"],
+        ),
+    )
+
+    def _run(self, text: str) -> str:
+        data = json.loads(text)
+        response = self.requests_wrapper.get(data["url"])
+        response = response[: self.response_length]
+        # print(f"Request tool input: {data}\nResponse: {response}")
+        return self.llm_chain.predict(
+            response=response, instructions=data["output_instructions"]
+        ).strip()
+
+    async def _arun(self, text: str) -> str:
+        pass
+
+
+class RequestsPostToolWithSummary(BaseRequestsTool, BaseTool):
+    """Tool for making a POST request to an API endpoint."""
+
+    name = "requests_post"
+    description = """Use this when you want to POST to a website.
+Input to the tool should be a json string with 3 keys: "url", "data", and "output_instructions".
+The value of "url" should be a string.
+The value of "data" should be a dictionary of key-value pairs you want to POST to the url.
+The value of "summary_instructions" should be instructions on what information to extract from the response, for example the id(s) for a resource(s) that the POST request creates.
+Always use double quotes for strings in the json string."""
+
+    response_length: Optional[int] = MAX_RESPONSE_LENGTH
+    llm_chain = llm_chain = LLMChain(
+        llm=DEFAULT_LLM,
+        prompt=PromptTemplate(
+            template="""Here is an API response:\n\n{response}\n\n====
+Your task is to extract some information according to these instructions: {instructions}
+When working with API objects, you should usually use ids over names. Do not return any ids or names that are not in the response.
+If the response indicates an error, you should instead output a summary of the error.
+
+Output:""",
+            input_variables=["response", "instructions"],
+        ),
+    )
+
+    def _run(self, text: str) -> str:
+        try:
+            data = json.loads(text)
+            response = self.requests_wrapper.post(data["url"], data["data"])
+            response = response[: self.response_length]
```

Comment:
Keeping the first `self.response_length` of the response looks super reasonable. I wonder if down the road there will be need for more sophisticated heuristics

Reply:
sadly the LM doesn't get full view of the response rn.. eg in my contrived example "get me the tracks from this album..." you can see in the PR description it only actually gets one track, as each track object is enormous